src/inline_tests.ml

```diff
@@ -56,32 +57,31 @@ module Backend = struct
       { info             : Info.t
       ; lib              : Lib.t
       ; runner_libraries : (Lib.t list, exn) result
-      ; extends          : (    t list, exn) result option
+      ; extends          : (    t list, exn) result
       }
 
     let desc ~plural = "inline tests backend" ^ if plural then "s" else ""
     let desc_article = "an"
 
     let lib  t = t.lib
-    let deps t = t.extends
+    let extends t = t.extends
 
     let instantiate ~resolve ~get lib (info : Info.t) =
       { info
       ; lib
-      ; runner_libraries = Result.all (List.map info.runner_libraries ~f:resolve)
+      ; runner_libraries =
+          Result.all (List.map info.runner_libraries ~f:resolve)
       ; extends =
           let open Result.O in
-          Option.map info.extends
-            ~f:(fun l ->
-              Result.all
-                (List.map l
-                   ~f:(fun ((loc, name) as x) ->
-                     resolve x >>= fun lib ->
-                     match get lib with
-                     | None ->
-                       Error (Loc.exnf loc "%S is not an %s" name
-                                (desc ~plural:false))
-                     | Some t -> Ok t)))
+          Result.all
+            (List.map info.extends
+               ~f:(fun ((loc, name) as x) ->
+                 resolve x >>= fun lib ->
+                 match get ~loc lib with
+                 | None ->
+                   Error (Loc.exnf loc "%S is not an %s" name
+                            (desc ~plural:false))
+                 | Some t -> Ok t))
```

Comment:
Would it be possible to move these changes into their own PR if possible?

Reply:
Ok, I'll cherry-pick directly some of the simpler commit to master directly, since we can't create dependent PRs on github

Reply:
Done: https://github.com/ocaml/dune/pull/632